commons/src/main/java/com/powsybl/commons/io/table/AbstractTableFormatter.java

```diff
@@ -26,12 +28,19 @@ protected AbstractTableFormatter(Locale locale, String invalidString) {
 
     protected abstract TableFormatter write(String value) throws IOException;
 
+    protected abstract TableFormatter write(String value, CellStyle cellStyle) throws IOException;
```

Comment:
You cannot expose a dependency to CellStyle. TableFormatter has to be fully independent of the chosen implementation. You have to create your own class, use it in the interface and convert it on the fly in the concrete class

Reply:
Your CellStyle class can handle the horizontal alignment and number formatting.

Reply:
_You cannot expose a dependency to CellStyle._ 
This is clear.
_TableFormatter has to be fully independent of the chosen implementation._ 
This is clear too.
_You have to create your own class, use it in the interface and convert it on the fly in the concrete class_
THis is not clear. Could you please be more specific ?